buf-perf.go

```diff
@@ -22,15 +24,14 @@ type PerfBuffer struct {
 	slot       uint
 	eventsChan chan []byte
 	lostChan   chan uint64
-	stop       chan struct{}
 	closed     bool
 	wg         sync.WaitGroup
+	stopFlag   atomic.Uint32 // atomic
```

Comment:
unit tests seem to fail on this.
You can define it as
`stopFlag   int32`
Then use
`atomic.StoreInt32(&pb.stopFlag, 1)`

It will also make the cast in the poll function more straight forward since it is an int and not atomic type:
(*C.int)(unsafe.Pointer(&pb.stopFlag))

Reply:
Unit tests fail for Go 1.18 since it was added in 1.19. So, we could deprecate 1.18.